src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_field.ts

```diff
@@ -549,7 +549,7 @@ export const useField = <T, FormType = FormData, I = T>(
       type,
       label,
       labelAppend,
-      helpText,
+      helpText: typeof helpText === 'function' ? helpText() : helpText,
```

Comment:
Here we will unwrap the lazy help text so that the useField() receives the `ReactNode` value. This allows to simplify bunch of existing places and not to fix bunch of other places where `helpText={helpText}` is already used